### PR TITLE
Debian patches

### PR DESCRIPTION
**Pull request by d-torrance**
Here is a series of patches present in the Debian package of wmsysmon (with some slight modifications).

Thanks for considering them!

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -1,7 +1,8 @@
 CC = gcc
+PREFIX = /usr/local
```
Comment (vcaputo) [inline src/Makefile]: For debian (or other distributions) does placement in usr/local make sense?

It made sense when all the users of wmsysmon were building from source, but if we're going down this road we may as well make it easier for distributors to place the binary in /usr/bin when it's managed by package management.  /usr/local/bin is typically reserved for locally installed packages, usually built from source by the sysadmin.

File: src/Makefile [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
 
 LIBDIR += -L/usr/X11R6/lib
-LDFLAGS += -lXpm -lXext -lX11 -lm -s
+LIBS = -lXpm -lXext -lX11 -lm
```
Comment (vcaputo) [inline src/Makefile]: while we're in here it probably makes sense to switch to using pkg-config for sourcing these flags.

e.g.
>LIBS=$(shell pkg-config --libs xpm xext x11) -lm

File: src/Makefile [hunk 3]
```diff
@@ -1,7 +1,8 @@
...
 #CFLAGS = -g #-DMONDEBUG
 CFLAGS += -Wall -O3 -fomit-frame-pointer -I/usr/X11R6/include #-DHI_INTS #undefine HI_INTS if your x86 SMP or alpha
```
Comment (vcaputo) [inline src/Makefile]: same for the include paths:
> pkg-config --cflags xpm xext x11
...
modern distros have dropped the X11 branches in the filesystem altogether, I'm ok with requiring pkg-config as a hard build dependency.

File: src/wmsysmon.1 [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+.TP
+.B \-h
+Show the help screen.
+.TP
+.B \-v
+Print wmsysmon's version number.
+
+.SH SEE ALSO
+.BR Website
+<ftp://gnugeneration.com/pub/Linux/wmsysmon>
```
Comment (vcaputo) [inline src/wmsysmon.1]: I haven't run an ftp service on the gnugeneration server for ages, we may as well just point things at the github url now.

File: src/wmsysmon.c [hunk 5]
```diff
@@ -850,10 +852,10 @@ void usage(void)
 #ifdef HI_INTS
 			"| 01234567   UV   |  <--- 0-N are hardware interrupts 0-23\n"
 			"| 89ABCDEF   WX   |  <--- U,V are Page IN/OUT, W,X are Swap IN/OUT\n"
-			"| GHIJKLMN   YZ   |\n"
+			"| GHIJKLMN   YZ   |  <--- Y,Z are Swap IN/OUT\n"
```
Comment (vcaputo) [inline src/wmsysmon.c]: I didn't look closely at the code yet, but YZ I think are just dead filler lights in the HI_INTS builds, so the current help is correct here.

**Comment by vcaputo** [comment]
Thanks for the contribution!  This project has been forgotten for a long time now.

Just a few minor things to fixup and I'll happily merge these changes.

The interrupt code also need reworking since modern systems aren't limited to the simple 16/24 interrupts split.  What should probably be happening nowadays is only put active interrupts into the visible set of interrupts, and include something like a mouseover to display the name of the driver servicing the interrupt, or something.

I also suspect the cpu/io/swap activity meters are broken on modern kernels.

File: src/Makefile (hunk 1 above)
Comment (d-torrance) [inline src/Makefile]: I can't speak for the other distributions, but in Debian we would definitely compile with PREFIX=/usr.  I'll make the change.

**Comment by d-torrance** [comment]
Thanks for the notes!  I'll send a revised pull request soon.

As far as things being broken on modern kernels, there's a 12-year-old Debian bug mentioning this [1].  I'll take a look.

[1] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=194214

**Comment by d-torrance** [comment]
I've addressed the comments.  Rather than adding a bunch of new commits, I figured I'd rewrite history in a new branch.  Please see #2.

Thanks!

**Pull request closed without merging**